### DISCUSSION TITLE
Best way to update?

### DISCUSSION THREAD

**Discussion by somewebmedia** [gregsullivan/_tw#92] [General]
Hello! First thanks for the awesome project!

You've made my life much easier with this! 🤗

So what would be the best way to be up to date with your new changes to the project? Since this is not a dependency that you call into your project, there is no way to automatically update it. What is your suggestion for this, to go by hand and copy the code? 😅

Thanks!

**Comment by gregsullivan** [comment]
Hello—Thanks for posting this!

_tw's philosophy in terms of updates is very much in line with [that of _s](https://themeshaper.com/2013/10/11/dont-update-your-theme/). Once you generate your theme, you shouldn't worry about keeping it in sync with every change from _tw.

That said, you may want, for example, to update to the latest version of Tailwind. Generally it's safe to use [`npm update`](https://docs.npmjs.com/cli/v8/commands/npm-update), though you'll always want to make sure you have a past commit you can go back to in case things go wrong. You could also copy, e.g., the most recent `devDependencies` from `package.json` and the most recent `package-lock.json` and then run `npm install`.

I tend only to do this if I'm going back to an older project and I specifically need a new Tailwind feature. I still have client sites on versions of _tw from two or three years ago, for whatever that's worth!

If there are bugs, or if something breaks for any reason, I generally create commits specific to fixing those bugs, and I'll create a list of steps to fix the bug in an existing theme. This has only really happened once, and you can see how that looks in [this issue](https://github.com/gregsullivan/_tw/issues/60).

Please let me know if you have any other questions!

**Reply by gregsullivan** [reply]
This has come up before, so I added a page specifically addressing this question to the documentation:

https://underscoretw.com/docs/updating/

Please let me know what you think!

**Comment by somewebmedia** [comment]
Thanks for the clarification. Yes, I ended up just copying the `devDependencies` from the `package.json` because that's what I needed for now.

**Reply by gregsullivan** [reply]
Perfect, happy to hear it!
